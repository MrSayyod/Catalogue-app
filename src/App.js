import './App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import Header from './components/Header';
import ProductListing from './components/ProductListing';
import ProductDetail from './components/ProductDetail';
import store from './store';

function App() {
  return (
    <div className="App">
      <Provider store={store}>
        <Router>
          <Header />
          <Switch>
            <Route path="/" exact component={ProductListing} />
            <Route path="/product/:productId" exact component={ProductDetail} />
            <Route>404 Not Found!</Route>
          </Switch>
        </Router>
      </Provider>
    </div>
  );
}

export default App;
