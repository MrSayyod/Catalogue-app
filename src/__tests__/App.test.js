import React from "react";
import App from "../App";
import { act } from "react-dom/test-utils";
import { render, unmountComponentAtNode } from "react-dom";
import pretty from "pretty";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders correctly", () => {
  act(() => {
    render(<App />, container);
  });
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
