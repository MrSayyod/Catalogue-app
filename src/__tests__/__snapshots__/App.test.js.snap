// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<div class=\\"App\\">
  <div class=\\"menu\\">
    <div class=\\"container\\">
      <h2>Heey header</h2>
    </div>
  </div>
  <div class=\\"menu_pdls\\">
    <div class=\\"category_filter\\"><select name=\\"category\\" class=\\"categoryDropdown\\">
        <option value=\\"ALL\\">CATEGORIES</option>
        <option valu=\\"men's clothing\\">men's clothing</option>
        <option valu=\\"jewelery\\">jewelery</option>
        <option valu=\\"electronics\\">electronics</option>
        <option valu=\\"women's clothing\\">women's clothing</option>
      </select></div>
    <div class=\\"container_pdls\\"></div>
    <section></section>
  </div>
</div>"
`;
